internal/backend/rest/rest.go

```diff
@@ -173,7 +174,29 @@ func (b *Backend) IsNotExist(err error) bool {
 // Load runs fn with a reader that yields the contents of the file at h at the
 // given offset.
 func (b *Backend) Load(ctx context.Context, h restic.Handle, length int, offset int64, fn func(rd io.Reader) error) error {
-	return backend.DefaultLoad(ctx, h, length, offset, b.openReader, fn)
+	r, err := b.openReader(ctx, h, length, offset)
+	if err != nil {
+		return err
+	}
+	err = fn(r)
```

Comment:
If `fn` returns an error, something else went wrong (and maybe the body wasn't even read completely), so we can skip waiting for EOF and just close the reader. Waiting for EOF only makes sense in the happy case when no error occurred:

```suggestion
	err = fn(r)
	if err != nil {
		_ = r.Close() // ignore error here
		return err
	}
```

This will also decouple the `err` variable from whatever happens below, which makes the code much easier to read.

Reply:
done